### PR TITLE
[release/2.8] fix miopen batchnorm changing output format

### PR DESCRIPTION
**Pull request by jeffdaily**
cherry pick of https://github.com/pytorch/pytorch/pull/162112

### CODE REVIEW

**Comment by jeffdaily** [comment]
depends on https://github.com/ROCm/pytorch/pull/2601, so merge that one first, then we'll need to fix conflicts.

**Comment by jithunnair-amd** [comment]
Also need to make sure this PR includes any fixes for https://github.com/pytorch/pytorch/pull/162112#issuecomment-3277205770

**Comment by jeffdaily** [comment]
Fixed new conflicts.  No other fixes needed.  Upstream PR is merged without changes.  Tests were flaky.

**Comment by jerrymannil** [comment]
! cherry-pick --onto release/2.9
